app/services/return-requirements/submit-purpose.service.js

```diff
@@ -69,8 +69,13 @@ async function _save (session, requirementIndex, payload) {
   return session.$update()
 }
 
-function _validate (payload) {
-  const validation = PurposeValidation.go(payload)
+async function _validate (payload, purposesData) {
+  const purposeIds = []
+  purposesData.forEach((purpose) => {
+    purposeIds.push(purpose.id)
+  })
```

Comment:
Sorry! Small NIT.

I think this scenario is what `map()` is idiomatically suited for.

```suggestion
  const purposeIds = purposesData.map((purpose) => {
    return purpose.id
  })
```